### PR TITLE
Sagemaker: add support for mme for neuron

### PR DESCRIPTION
**Pull request by aws-vrnatham**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/inference/docker/build_artifacts/sagemaker/python_service.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 TFS_REST_PORT_RANGE = os.environ.get("TFS_REST_PORT_RANGE")
 SAGEMAKER_TFS_PORT_RANGE = os.environ.get('SAGEMAKER_SAFE_PORT_RANGE')
 TFS_INSTANCE_COUNT = int(os.environ.get("SAGEMAKER_TFS_INSTANCE_COUNT", "1"))
+INFERENTIA_DEVICES = os.environ.get("INFERENTIA_DEVICES")
```
Comment (micwade-aws) [inline tensorflow/inference/docker/build_artifacts/sagemaker/python_service.py:38]: Let's change this to NEURON_DEVICES to be consistent?

File: tensorflow/inference/docker/build_artifacts/sagemaker/serve.py [hunk 2]
```diff
@@ -103,6 +107,7 @@ def __init__(self):
             # set environment variable for python service
             os.environ["TFS_GRPC_PORT_RANGE"] = self._tfs_grpc_port_range
             os.environ["TFS_REST_PORT_RANGE"] = self._tfs_rest_port_range
+        self._inferentia_devices = find_inferentia_devices('/dev', 'neuron*')
```
Comment (micwade-aws) [inline tensorflow/inference/docker/build_artifacts/sagemaker/serve.py:110]: _neuron_devices = find_neuron_devices()

**Review by micwade-aws — CHANGES_REQUESTED** [review]
Naming changes requested to refer to the devices as neuron devices instead of inferentia.  The commit history is a bit odd as well - recommend cleaning up the commits before submitting the PR.

**Comment by arjkesh** [comment]
@aws-vrnatham - thanks for the PR! Please let us know if this is in active development or can be closed. If ready for review, please move from "Draft" to "Open" and we'll take a look. Thanks!

**Pull request closed without merging**